instrumentor/controllers/sourceinstrumentation/namespace_controller.go

```diff
@@ -44,18 +44,21 @@ func (r *NamespaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, err
 	}
 
+	var reconcileFunc reconcileFunction
 	enabled, _, err := sourceutils.IsObjectInstrumentedBySource(ctx, r.Client, &ns)
```

Comment:
If we got `NotFound` when trying to "Get" the ns object, the `if` 5 lines above will not return from this function, and we will be using an uninitialized object here.

I think it should be changed to:

```go
if err != nil {
  return ctrl.Result{}, client.IgnoreNotFound(err)
}
```

(also no need to print to log as the controller runtime will do that for us if we return err from the reconcile, which will end up with 2 prints for the same message in the log)